src/decisionengine/framework/logicengine/BooleanExpression.py

```diff
@@ -45,23 +51,28 @@ def function_name_from_call(callnode):
 
 class BooleanExpression:
     def __init__(self, expr):
-        self.expr_str = expr
+        self.expr_str, self.fail_on_error = maybe_fail_on_error(str.strip(expr))
         source = 'string'
         mode = 'eval'
-        syntax_tree = ast.parse(expr, source, mode)
+        syntax_tree = None
+        try:
+            syntax_tree = ast.parse(self.expr_str, source, mode)
+        except Exception:
```

Comment:
Since this is a defensive exception, is there a way to trigger it or should we add `# pragma: no cover` to note the purpose?

Reply:
Yes, pretty sure we can trigger the exception.  I'll attempt to do so.

Reply:
See most recent push.